File: 이진영/Graph/순위.js
function solution(n, results) {
    let graph = Array.from(Array(n), () => Array(n).fill(0));
    results.forEach(val => {
        graph[val[0] - 1][val[1] - 1] = 1;
        graph[val[1] - 1][val[0] - 1] = -1;
    })
    
    for (let k = 0; k < n; k++) {
        for (let i = 0; i < n; i++) {
            for (let j = 0; j < n; j++) {
                if (i === j) {
                    graph[i][j] = 0;
                    continue;
                }
                if (graph[i][k] > 0 && graph[k][j] > 0) graph[i][j] = 1;
                else if (graph[i][k] < 0 && graph[k][j] < 0) graph[i][j] = -1;
            }
        }
    }
    
    return n - graph.filter(i => i.filter(i => i === 0).length > 1).length
}

File: 김채린/Graph/순위.kt
class Solution {
fun solution(
    n: Int,
    results: Array<IntArray>,
): Int {
    var resultArr = Array(n + 1) { IntArray(n + 1){2} }
    for (result in results) {
        resultArr[result[0]][result[1]] = 1
        resultArr[result[1]][result[0]] = -1
    }
    for (k in 1..n) {
        for (i in 1..n) {
            for (j in 1..n) {
                if (resultArr[i][j] == 2) {
                    if (i==j){
                        resultArr[i][j] = 0
                        continue
                    }
                    if (resultArr[i][k] == 1 && resultArr[k][j] == 1) {
                        resultArr[i][j] = 1
                    }
                    if (resultArr[i][k] == -1 && resultArr[k][j] == -1) {
                        resultArr[i][j] = -1
                    }
                }
            }
        }
    }
    var result = n
    for (i in 1..n) {
        for (j in 1..n) {
            if (resultArr[i][j] == 2) {
                result--
                break
            }
        }
    }
    return result
}

}
